### DISCUSSION TITLE
Unable to create a FirestoreClient: Error occurred during parsing: Fail to push limit.

### DISCUSSION THREAD

**Discussion by vincenttarrit** [kreait/firebase-php#572] [Q&A]
Hello, 

I'm playing with this dependencies and in dev it's amazing. However, when I try in prod on my server I get this error: 

Unable to create a FirestoreClient: Error occurred during parsing: Fail to push limit.

on the line : vendor/kreait/firebase-php/src/Firebase/Factory.php:553

Do you know what is the problem and how I could fix it ? 

Thank you so much !

**Comment by jeromegamez** [comment]
I haven't encountered this error before, but it since it happens during the creation of the FirsetoreClient, the only thing that comes to my mind is that the ServiceAccount file/JSON could not be correctly parsed. 

This you can check by trying to make a connection through another component of the SDK, e.g. with `$factory->createDatabase()->getReference('/')->getValue()` - if that gives you something (or even a Not Found Error), the credentials file should be alright. Although the Firestore Client would have been more clear if it really was the credentials, I think... 🤔

Another approach would be to test if it works in another remote environment/another server - if it doesn't, try checking if the Firestore API is enabled in your project's Google Cloud Console. 🤞

**Comment by luisaureliocasoni** [comment]
I have the same problem. 

I have implanted the Firebase SDK in my production server, with PHP 7.4 installed. The error `Unable to create a FirestoreClient: Error occurred during parsing: Fail to push limit.` shows at the line `$firestore = $this->factory->createFirestore();`, after the successul connection.

I have translated the JSON key on my development server, with PHP 8 installed, and works fine, using the same JSON private key file located on my production server. Now, the firestore configuration is not the problem.

Did have other idea to attempt to resolve this problem?

**Reply by luisaureliocasoni** [reply]
I have checked the gRPC extension version on my production server. The version installed is 1.12.0, but the last version is 1.36.0.

I have pleadged to the hosting provider to upgrade the gRPC version, but they have negated my order. 

I will change the Firestore client to calls to direct Firestore REST API. 

Thanks.
